### ISSUE TITLE
V.1.1.1 won't start

### ISSUE THREAD

**Issue by fabiofidanza** [AkdM/KnightTouchBar2000#9]
I get an "App quit unexpectedly" error.

Process:               KnightTouchBar2000 [3840]
Path:                  /private/var/folders/*/KnightTouchBar2000.app/Contents/MacOS/KnightTouchBar2000
Identifier:            com.AnthonyDaMota.KnightTouchBar2000
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           KnightTouchBar2000 [3840]
User ID:               501

Date/Time:             2018-07-31 22:13:41.431 +0200
OS Version:            Mac OS X 10.13.6 (17G2112)
Report Version:        12
Bridge OS Version:     2.4.1 (15P6613)
Anonymous UUID:        165CDA4F-F2DE-6C20-E2BC-11DF9D097383

Sleep/Wake UUID:       56E62EDD-91E5-45BF-916C-0F5EBFED3968

Time Awake Since Boot: 20000 seconds
Time Since Wake:       1500 seconds

System Integrity Protection: enabled

Notes:                 Translocated Process

Crashed Thread:        0

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

VM Regions Near 0 (cr2):
--> 
    __TEXT                 000000010825a000-0000000108262000 [   32K] r-x/rwx SM=COW  v []

Thread 0 Crashed:
0   ???                           	0x000000010e80d19c _dyld_start + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffee79a5a48
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x000000010e80d19c  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0

Binary Images:
       0x10825a000 -        0x108261fff +??? (0) <247B3159-36C6-3833-83BC-D5621A8DBD8C> (null)
       0x10e80c000 -        0x10e856acf +??? (551.4) <8A72DE9C-A136-3506-AA02-4BA2B82DCAF3> (null)

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 517732
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=464K resident=0K(0%) swapped_out_or_unallocated=464K(100%)
Writable regions: Total=8404K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8404K(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
__DATA                             236K        4 
__LINKEDIT                         132K        3 
__TEXT                             332K        3 
shared memory                        8K        3 
===========                     =======  ======= 
TOTAL                             64.7M       11 

Model: MacBookPro15,1, BootROM 15.16.6613.0.0,0, 6 processors, Intel Core i7, 2,6 GHz, 32 GB, SMC 
Graphics: kHW_IntelUHDGraphics630Item, Intel UHD Graphics 630, Built-In
Graphics: kHW_AMDRadeonPro560XItem, Radeon Pro 560X, PCIe
Memory Module: BANK 0/ChannelA-DIMM0, 16 GB, DDR4, 2400 MHz, Micron, 16ATS2G64HZ-2G6B1 16ATS2G64HZ-2G6B1
Memory Module: BANK 2/ChannelB-DIMM0, 16 GB, DDR4, 2400 MHz, Micron, 16ATS2G64HZ-2G6B1 16ATS2G64HZ-2G6B1
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x7BF), wl0: Jun 24 2018 21:16:16 version 9.130.28.44.32.1.55 FWID 01-55577fa3
Bluetooth: Version 6.0.7f10, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB 3.1 Bus
USB Device: iBridge Bus
USB Device: iBridge DFR brightness
USB Device: iBridge Display
USB Device: Apple Internal Keyboard / Trackpad
USB Device: Headset
USB Device: iBridge ALS
USB Device: iBridge FaceTime HD Camera (Built-in)
USB Device: iBridge
Thunderbolt Bus: MacBook Pro, Apple Inc., 28.10
Thunderbolt Bus: MacBook Pro, Apple Inc., 28.10

**Comment by andyWynyard** [comment]
Yep, I get an issue also:
```Process:               KnightTouchBar2000 [13861]
Path:                  /Applications/KnightTouchBar2000.app/Contents/MacOS/KnightTouchBar2000
Identifier:            com.AnthonyDaMota.KnightTouchBar2000
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           KnightTouchBar2000 [13861]
User ID:               501

Date/Time:             2018-11-02 11:51:10.609 +0100
OS Version:            Mac OS X 10.14.1 (18B75)
Report Version:        12
Bridge OS Version:     3.0 (14Y667)
Anonymous UUID:        369773A5-3F92-A684-DDCE-7E9373D9F083

Sleep/Wake UUID:       378ED95F-7379-42AB-B1B1-D6980CDDC5C4

Time Awake Since Boot: 78000 seconds
Time Since Wake:       2000 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

VM Regions Near 0 (cr2):
--> 
    __TEXT                 0000000103e14000-0000000103e1c000 [   32K] r-x/rwx SM=COW  

Thread 0 Crashed:
0                                 	0x0000000109154000 _dyld_start + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffeebdebb88
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x0000000109154000  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0


Binary Images:
       0x103e14000 -        0x103e1bfff + (0) <247B3159-36C6-3833-83BC-D5621A8DBD8C> 
       0x109153000 -        0x1091d16a7 + (635.2) <1780094A-8FE2-3EAA-B4A3-C4CF14BC5196> 

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 66188
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=728K resident=0K(0%) swapped_out_or_unallocated=728K(100%)
Writable regions: Total=8400K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8400K(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
__DATA                             240K        4 
__LINKEDIT                         188K        3 
__TEXT                             540K        3 
shared memory                        8K        3 
===========                     =======  ======= 
TOTAL                             65.0M       11 

Model: MacBookPro14,3, BootROM 185.0.0.0.0, 4 processors, Intel Core i7, 2.9 GHz, 16 GB, SMC 2.45f0
Graphics: Intel HD Graphics 630, Intel HD Graphics 630, Built-In
Graphics: Radeon Pro 560, Radeon Pro 560, PCIe
Memory Module: BANK 0/DIMM0, 8 GB, LPDDR3, 2133 MHz, 0x802C, 0x4D5435324C31473332443450472D30393320
Memory Module: BANK 1/DIMM0, 8 GB, LPDDR3, 2133 MHz, 0x802C, 0x4D5435324C31473332443450472D30393320
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x173), Broadcom BCM43xx 1.0 (7.77.61.1 AirPortDriverBrcmNIC-1305.2)
Bluetooth: Version 6.0.9f2, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB 3.0 Bus
USB Device: iBridge
Thunderbolt Bus: MacBook Pro, Apple Inc., 39.2
Thunderbolt Bus: MacBook Pro, Apple Inc., 39.2
```

**Comment by andyWynyard** [comment]
This happened to me also. Wish I had the time to investigate.

**Comment by AkdM** [comment]
Sorry for the issue guys. I am aware of it.
I think this is due to the fact the app isn't signed and that I don't have any certificate, but I didn't investigate further.
As a fallback, you can still compile it yourselves and it will work nicely.

**Comment by marchnic** [comment]
Hi,
sorry if my question his stupid, but how do you compile the project ? I am totally newbie...
Thanks

**Comment by 956MB** [comment]
> I get an "App quit unexpectedly" error.
> 
> Process: KnightTouchBar2000 [3840]
> Path: /private/var/folders/*/KnightTouchBar2000.app/Contents/MacOS/KnightTouchBar2000
> Identifier: com.AnthonyDaMota.KnightTouchBar2000
> Version: ???
> Code Type: X86-64 (Native)
> Parent Process: ??? [1]
> Responsible: KnightTouchBar2000 [3840]
> User ID: 501
> 
> Date/Time: 2018-07-31 22:13:41.431 +0200
> OS Version: Mac OS X 10.13.6 (17G2112)
> Report Version: 12
> Bridge OS Version: 2.4.1 (15P6613)
> Anonymous UUID: 165CDA4F-F2DE-6C20-E2BC-11DF9D097383
> 
> Sleep/Wake UUID: 56E62EDD-91E5-45BF-916C-0F5EBFED3968
> 
> Time Awake Since Boot: 20000 seconds
> Time Since Wake: 1500 seconds
> 
> System Integrity Protection: enabled
> 
> Notes: Translocated Process
> 
> Crashed Thread: 0
> 
> Exception Type: EXC_CRASH (Code Signature Invalid)
> Exception Codes: 0x0000000000000000, 0x0000000000000000
> Exception Note: EXC_CORPSE_NOTIFY
> 
> Termination Reason: Namespace CODESIGNING, Code 0x1
> 
> kernel messages:
> 
> VM Regions Near 0 (cr2):
> -->
> __TEXT 000000010825a000-0000000108262000 [ 32K] r-x/rwx SM=COW v� []
> 
> Thread 0 Crashed:
> 0 ??? 0x000000010e80d19c _dyld_start + 0
> 
> Thread 0 crashed with X86 Thread State (64-bit):
> rax: 0x0000000000000000 rbx: 0x0000000000000000 rcx: 0x0000000000000000 rdx: 0x0000000000000000
> rdi: 0x0000000000000000 rsi: 0x0000000000000000 rbp: 0x0000000000000000 rsp: 0x00007ffee79a5a48
> r8: 0x0000000000000000 r9: 0x0000000000000000 r10: 0x0000000000000000 r11: 0x0000000000000000
> r12: 0x0000000000000000 r13: 0x0000000000000000 r14: 0x0000000000000000 r15: 0x0000000000000000
> rip: 0x000000010e80d19c rfl: 0x0000000000000200 cr2: 0x0000000000000000
> 
> Logical CPU: 0
> Error Code: 0x00000000
> Trap Number: 0
> 
> Binary Images:
> 0x10825a000 - 0x108261fff +??? (0) <247B3159-36C6-3833-83BC-D5621A8DBD8C> (null)
> 0x10e80c000 - 0x10e856acf +??? (551.4) <8A72DE9C-A136-3506-AA02-4BA2B82DCAF3> (null)
> 
> External Modification Summary:
> Calls made by other processes targeting this process:
> task_for_pid: 0
> thread_create: 0
> thread_set_state: 0
> Calls made by this process:
> task_for_pid: 0
> thread_create: 0
> thread_set_state: 0
> Calls made by all processes on this machine:
> task_for_pid: 517732
> thread_create: 0
> thread_set_state: 0
> 
> VM Region Summary:
> ReadOnly portion of Libraries: Total=464K resident=0K(0%) swapped_out_or_unallocated=464K(100%)
> Writable regions: Total=8404K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8404K(100%)
> 
> ```
>                             VIRTUAL   REGION 
> ```
> REGION TYPE SIZE COUNT (non-coalesced)
> =========== ======= =======
> STACK GUARD 56.0M 2
> Stack 8192K 2
> __DATA 236K 4
> __LINKEDIT 132K 3
> __TEXT 332K 3
> shared memory 8K 3
> =========== ======= =======
> TOTAL 64.7M 11
> 
> Model: MacBookPro15,1, BootROM 15.16.6613.0.0,0, 6 processors, Intel Core i7, 2,6 GHz, 32 GB, SMC
> Graphics: kHW_IntelUHDGraphics630Item, Intel UHD Graphics 630, Built-In
> Graphics: kHW_AMDRadeonPro560XItem, Radeon Pro 560X, PCIe
> Memory Module: BANK 0/ChannelA-DIMM0, 16 GB, DDR4, 2400 MHz, Micron, 16ATS2G64HZ-2G6B1 16ATS2G64HZ-2G6B1
> Memory Module: BANK 2/ChannelB-DIMM0, 16 GB, DDR4, 2400 MHz, Micron, 16ATS2G64HZ-2G6B1 16ATS2G64HZ-2G6B1
> AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x7BF), wl0: Jun 24 2018 21:16:16 version 9.130.28.44.32.1.55 FWID 01-55577fa3
> Bluetooth: Version 6.0.7f10, 3 services, 27 devices, 1 incoming serial ports
> Network Service: Wi-Fi, AirPort, en0
> USB Device: USB 3.1 Bus
> USB Device: iBridge Bus
> USB Device: iBridge DFR brightness
> USB Device: iBridge Display
> USB Device: Apple Internal Keyboard / Trackpad
> USB Device: Headset
> USB Device: iBridge ALS
> USB Device: iBridge FaceTime HD Camera (Built-in)
> USB Device: iBridge
> Thunderbolt Bus: MacBook Pro, Apple Inc., 28.10
> Thunderbolt Bus: MacBook Pro, Apple Inc., 28.10

I get this same error message when I try to run the .app, as well as these errors in Xcode when I try to build and run. 
<img width="1552" alt="screen shot 2018-12-05 at 1 11 36 am" src="https://user-images.githubusercontent.com/39519792/49496285-22725300-f82b-11e8-95fc-941890b9f214.png">

Sorry if this is a lot. I am completely new to Xcode and Swifts workings, so I dont what to do with these errors  here to make it work. Just pointing it out. :)

**Comment by AkdM** [comment]
@Bloumbs You simply need to add your developper account to Xcode so you can build.

`There are no accounts registered with Xcode. Add your developer account to Xcode [...]`

**Comment by 956MB** [comment]
@AkdM I don't have an Apple developer account. Im only half a year into programming and haven't tried anything with Swift & Xcode. Do you have plans of possibly fixing the .app so people can use it more easily? Or is there something else I can try to build and run it?

**Comment by AkdM** [comment]
@Bloumbs You simply need to login with your Apple ID under the General settings of the target > Signing > Add Account. It does not require a paid developer account. Even if I fix the .app, the same error will happen someday. 🙂

<img width="1552" alt="screen shot 2018-12-09 at 12 37 06" src="https://user-images.githubusercontent.com/4175461/49696758-344b5180-fbaf-11e8-9125-d29304b319ec.png">

**Comment by Bloumbs** [comment]
@AkdM Signing in like you said got it to build and run. But after that I got this output:  and nothing happened after that. Should I next convert to Swift 4.2 like the error says?  I wanted to ask first before I tried, because all of this is pretty above my head. :) 
<img width="1552" alt="screen shot 2018-12-09 at 6 35 01 am" src="https://user-images.githubusercontent.com/39519792/49697420-bf5f2400-fb7c-11e8-9fe2-6d839b1d468b.png">
